S2U-27 Messages: Convert a private message into a forum message in the FAQ forum

stetsche opened the pull request:
https://sakaiproject.atlassian.net/browse/SAK-35033

Conversion scripts: https://github.com/sakaiproject/sakai-reference/pull/204

jesusmmp commented on msgcntr/messageforums-app/pom.xml:
```diff
@@ -88,6 +88,10 @@
 			<groupId>org.springframework</groupId>
 			<artifactId>spring-core</artifactId>
 		</dependency>
+        <dependency>
```
Spacing

jesusmmp commented on msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/DiscussionForumTool.java:
```diff
@@ -8138,6 +8153,10 @@ public String getEditorRows() {
 	}
 
+	public String getEditorCols() {
+		return "132";
```
We should avoid magic numbers ;)

jesusmmp commented on msgcntr/messageforums-component-impl/src/java/org/sakaiproject/component/app/messageforums/AreaManagerImpl.java:
```diff
@@ -48,6 +48,7 @@
 import org.sakaiproject.tool.api.Placement;
 import org.sakaiproject.tool.api.SessionManager;
 import org.sakaiproject.tool.api.ToolManager;
+import org.sakaiproject.user.api.UserDirectoryService;
```
Undo this change and line 223. No changes on this file

jesusmmp commented on msgcntr/messageforums-component-impl/src/java/org/sakaiproject/component/app/messageforums/MessageForumsForumManagerImpl.java:
```diff
@@ -710,6 +733,58 @@ public ActorPermissions createDefaultActorPermissions()
+    private DiscussionForum createFaqForum(Area discussionArea) {
+        log.debug("Creating a new FAQ Forum");
+
+        // Locale locale = Locale.getDefault();
```
Remove comment

jesusmmp commented on msgcntr/messageforums-hbm/src/java/org/sakaiproject/component/app/messageforums/dao/hibernate/Topic.hbm.xml:
```diff
@@ -191,6 +191,9 @@
 	    <property name="restrictPermissionsForGroups">
 	      <column name="RESTRICT_PERMS_FOR_GROUPS" not-null="true" default="0" />
 	    </property>
+      <property name="faqTopic">
```
Spacing

mpellicer replied:
Spacing vs MSGCNTR, name a better DUO!

stetsche replied:
Yes, I love it when i have to mix tabs with spaces to "fix" the indentation xD

mpellicer commented:
Thanks for helping me with the reviews @jesusmmp !

stetsche commented:
@jesusmmp @mpellicer Thanks for taking a look! I addressed the issues together with a few other minor changes﻿.
I also added the translations that @NuriaQS and @DavidVegaSa kindly provided.
I think the PR is ready now!

stetsche commented:
Hi @PabloSanRoman﻿, could you please help with the following translations?

```
pvt_publish_to_faq=Publish to FAQ
pvt_title=Title
pvt_question=Question
pvt_question_title_prefix=Q:
pvt_answer=Answer
pvt_answer_title_prefix=A:
pvt_answer_info=You can leave the answer field empty if you don\u0027t want to post it yet.
pvt_missing_title=You must enter a title before you can post to FAQ.
pvt_missing_question=You must enter a question before you can post to FAQ.

cdfm_faq_forum_title=FAQ Forum
cdfm_faq_forum_description=Forum for frequently asked questions
cdfm_faq_topic_title=FAQ Topic
cdfm_faq_topic_description=Topic for frequently asked questions
```

Greetings,
Markus

jesusmmp commented on msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/DiscussionForumTool.java:
```diff
@@ -4285,11 +4287,50 @@ public String processDfMsgReplyMsgFromEntire()
+
+    if (StringUtils.startsWith(title, replyPrefix)) {
+        // Re: title --> Re: title
+        log.debug("LOL 1.0");
```
Remove this one and the others LOL xD

stetsche replied:
:sweat_smile:

jesusmmp commented on msgcntr/messageforums-app/src/webapp/jsp/privateMsg/pvtMsgPublishToFaqEdit.jsp:
```diff
@@ -0,0 +1,90 @@
+<%-- TODO Master merge:
```
Great annotations!

jesusmmp commented on samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/evaluation/QuestionScoresBean.java:
```diff
@@ -182,6 +182,9 @@ public class QuestionScoresBean implements Serializable, PhaseAware {
   private boolean emiItemPresent;
 
+  @Setter @Getter
+  private boolean randomItemPresent;
```
This code is not related to this patch.

jesusmmp commented on samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/util/ItemCancellationUtil.java:
```diff
@@ -17,46 +17,38 @@
 package org.sakaiproject.tool.assessment.util;
 
 import org.sakaiproject.tool.assessment.data.ifc.assessment.ItemDataIfc;
+import org.sakaiproject.tool.assessment.data.ifc.assessment.SectionDataIfc;
```
Review all the samigo code. Not related to this feature.

stetsche commented:
Thanks @jesusmmp﻿, done!

stetsche commented:
Codacy issues seem like all false-positives to me.

mpellicer commented on msgcntr/messageforums-api/src/bundle/org/sakaiproject/api/app/messagecenter/bundle/Messages_eu.properties:
```diff
@@ -891,3 +891,19 @@ perm-msg.permissions.allowToField.users=Pertsona guztiei bidali
 perm-msg.permissions.allowToField.myGroupRoles=Norberaren taldeen rolei bidali
+
+#S2U-27
+pvt_publish_to_faq=Publish to FAQ
```
Hey @pablosanroman, we have a small number of new keys, would you mind translating them? Not in a hurry.

PabloSanRoman replied:
Hi @mpellicer!here it is!: 

pvt_title=Titulua
pvt_question=Galdera
pvt_question_title_prefix=Q:
pvt_answer=Erantzuna
pvt_answer_title_prefix=A:
pvt_answer_info=Erantzunaren eremua hutsik utz dezakezu publikatu nahi ez baduzu oraindik. 
pvt_missing_title=Ezinbestekoa da izenburu bat sartzea "Ohiko galderak" atalean publikatzeko. 
pvt_missing_question=Ezinbestekoa da galdera bat sartzea "Ohiko galderak" atalean publikatzeko. 

cdfm_faq_forum_title=FAQ foroa
cdfm_faq_forum_description=Foroa Ohiko galderentzat
cdfm_faq_topic_title=FAQ Gaia
cdfm_faq_topic_description=Gaia Ohiko galderentzat

mpellicer commented:
This looks great, good job monster @stetsche ! I'm missing the conversion scripts, would you mind contributing and linking them?
